neuralop/layers/padding.py

```diff
@@ -52,33 +57,50 @@ def pad(self, x):
 
         except KeyError:
             padding = [int(round(p*r)) for (p, r) in zip(self.domain_padding, resolution)]
-            
+
             print(f'Padding inputs of {resolution=} with {padding=}, {self.padding_mode}')
-            
-            
+
+            # padding is being applied in reverse order (so we must reverse the padding list)
+            padding = padding[::-1]  
+
             output_pad = padding
 
             output_pad = [int(round(i*j)) for (i,j) in zip(self.output_scaling_factor,output_pad)]
-            
-            
+                        
             # the F.pad(x, padding) funtion pads the tensor 'x' in reverse order of the "padding" list i.e. the last axis of tensor 'x' will be
             # padded by the amount mention at the first position of the 'padding' vector.
             # The details about F.pad can be found here : https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html
 
             if self.padding_mode == 'symmetric':
                 # Pad both sides
-                unpad_indices = (Ellipsis, ) + tuple([slice(p, -p, None) for p in output_pad[::-1] ])
+                unpad_list = list()
+                for p in output_pad[::-1]:
+                    if p == 0:
+                        unpad_amount_pos = None
+                        unpad_amount_neg = None
+                    else:
+                        unpad_amount_pos = p
+                        unpad_amount_neg = -p
+                    unpad_list.append(slice(unpad_amount_pos, unpad_amount_neg, None))
```

Comment:
I actually don't mind spelling it out but can we call it padding_start and padding_end or something clearer? :)

Reply:
name modified :)